### DISCUSSION TITLE
Should hook useToggle ignore React events?

### DISCUSSION THREAD

**Discussion by ruggi99** [react-hookz/web#696] [Q&A] [answered]
Hi all,
I tried to use `useToggle` on a `<button>` to toggle a menu but it doesn't work because when used directly in a `<button>` the first argument is always the event fired.
```js
function MyComponent() {
  const [state, toggleState] = useToggle(false);
  return <button onClick={toggleState}>Toggle me!<button>
}
```

I'm wondering if we can ignore the first argument if it is an event (or a click event in this case).
The workaround for now is using `onClick={() => toggleState()}` but is not stable across renders.
Thanks

**Comment by AndreasNel** [comment, marked as answer]
Hi there!

This functionality was introduced in #867 and released in version [15.0.0](https://github.com/react-hookz/web/releases/tag/v15.0.0).
